netbox/extras/reports.py

```diff
@@ -22,23 +23,18 @@ def get_report(module_name, report_name):
     """
     Return a specific report from within a module.
     """
-    file_path = '{}/{}.py'.format(settings.REPORTS_ROOT, module_name)
+    reports = get_reports()
+    for grouping, report_list in reports:
+        if grouping != module_name:
+            continue
+        for report in report_list:
+            if report.name == report_name:
+                return report
 
-    spec = importlib.util.spec_from_file_location(module_name, file_path)
-    module = importlib.util.module_from_spec(spec)
-    try:
-        spec.loader.exec_module(module)
-    except FileNotFoundError:
-        return None
+    return None
 
-    report = getattr(module, report_name, None)
-    if report is None:
-        return None
 
-    return report()
-
```

Comment:
An explanatory note: this is a rework of the `get_report` method as the prior implementation would only work for local reports (as it was hardcoded to import relative to the `REPORTS_ROOT` directory). Instead of re-importing and re-instantiating the requested `Report` subclass, this now takes advantage of the fact that `get_reports()` already returns a dict of lists of `Report` subclass instances, and simply locates and returns the desired instance. This works for both local and plugin-packaged reports as `get_reports()` already handles both.